extensions/interactions/NumericInput/NumericInput.js

```diff
@@ -24,21 +24,36 @@ oppia.directive('oppiaInteractiveNumericInput', [
   'oppiaHtmlEscaper', function(oppiaHtmlEscaper) {
     return {
       restrict: 'E',
-      scope: {},
+      scope: {
+        getAnswerClassificationData: '&answerClassificationData'
+      },
       templateUrl: 'interaction/NumericInput',
-      controller: ['$scope', '$attrs', 'focusService',
-          function($scope, $attrs, focusService) {
+      controller: ['$scope', '$attrs', 'focusService', 'numericInputService',
+          function($scope, $attrs, focusService, numericInputService) {
         $scope.answer = '';
         $scope.labelForFocusTarget = $attrs.labelForFocusTarget || null;
+        var answerClassificationData = $scope.getAnswerClassificationData();
 
         $scope.NUMERIC_INPUT_FORM_SCHEMA = {
           type: 'float',
           ui_config: {}
         };
 
+        var USE_FRONTEND_RULE_EVALUATION = true;
+
         $scope.submitAnswer = function(answer) {
-          if (answer !== undefined && answer !== null) {
-            $scope.$parent.$parent.submitAnswer(Number(answer));
+          if (answer !== undefined && answer !== null && answer !== '') {
```

Comment:
I also agree with moving this to a central location (I'm not sure if you addressed this in your reply). It will simplify calling the post-classification functions, as well (avoids duplicating those calls across all of the interactions).